source/grid/grid_out.cc

```diff
@@ -3235,12 +3242,10 @@ GridOut::write_mesh_per_processor_as_vtu(
           patch.vertices[vertex] = cell->vertex(vertex);
           patch.data(0, vertex)  = cell->level();
           if (!cell->has_children())
-            patch.data(1, vertex) =
-              (double)static_cast<int>(cell->subdomain_id());
+            patch.data(1, vertex) = static_cast<int>(cell->subdomain_id());
```

Comment:
here and a couple lines down, the cast to an integer type is no longer necessary. but I'd like to retain the cast to `double`

Reply:
OK, your choice then. I would just keep both then, and maybe also cast to `signed int` to just make things clear.